R/tribble.R

```diff
@@ -19,6 +30,26 @@
 #'   "c",   3
 #' )
 #'
+#' # Conversion expressions can be supplied, including additional
+#' # paremeters if needed.
+#' tribble(
+#'   ~colA, colB~factor(., levels = c("B", "A")),
```

Comment:
Regarding the first point, I think I have a solution. `match.call()` allows the entire unevaluated parse tree for the call to be retrieved, complete with the expanded `...`. I've prototyped some code which handles `colA=, colB = lubridate(ymd(.)), ~colC`.

Regarding `magrittr`-style code, one option would be to check for a placeholder, and if none is found place it as the first argument of the first `call` that does not have a `call` as its first argument. That should work as expected, though it will mean code cannot be reused from `magrittr` but will need to be specific to `tribble`.